metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -721,3 +717,40 @@ def validate_token(name, token_prefix, authorize, instructions_fn=None):
 
     store_token(token_prefix, token)
     return True
+
+
+def validate_run_id(workflow_name, run_id):
+    """
+    Validates that a run_id adheres to the Argo Workflows naming rules, and
+    that it belongs to the current flow (accounting for project branch as well).
+    """
+    # Verify that user is trying to change an Argo workflow
+    if not run_id.startswith("argo-"):
+        raise MetaflowException("Argo workflow execution id's start with 'argo-'")
+
+    # Verify that run_id belongs to the Flow, and that branches match
+    name = run_id[5:]
+    workflow = ArgoWorkflows.get_execution(name)
+    if workflow is None:
+        raise MetaflowException("Could not find the workflow *%s*" % run_id)
+
+    _owner, _token, deployed_flow_name, deployed_workflow_template_name = workflow
+
+    # Verify we are operating on the correct Flow file compared to the running one.
+    # Without this check, using --name could be used to run commands for arbitrary run_id's, disregarding the Flow in the file.
+    if current.flow_name != deployed_flow_name:
+        raise MetaflowException(
+            "The workflow with the run_id *%s* is not for the flow *%s*.\n"
+            "Please use the correct Flow file." % (run_id, current.flow_name)
+        )
+
+    # Verify that we are operating on the correct project and branch.
+    # Project and branch are stored in the workflow template name so this is the easiest comparison to perform.
+    if workflow_name != deployed_workflow_template_name:
+        raise MetaflowException(
+            "The run_id *%s* belongs to the workflow *%s*, not to the workflow *%s*\n"
```

Comment:
We don't communicate to the user how `deployed_workflow_template_name` is imputed. Can we check for `project_name` and `branch_name` and compare them with that?

Reply:
Added some more robust checks that separately verify that the user is targeting the correct branch and project, and notifying which one is not matching the deployed workflow. Do the changes seem adequate?